components/net/http_loader.rs

```diff
@@ -1417,6 +1420,35 @@ fn http_network_or_cache_fetch(
         *done_chan = None;
     }
 
+    fn prompt_user_for_credentials(
+        context: &FetchContext,
+    ) -> std::option::Option<std::string::String> {
+        let local_embedder_proxy: EmbedderProxy;
+        {
+            local_embedder_proxy = context.state.embedder_proxy.lock().unwrap().clone();
+        }
```

Comment:
We can do `let embedder_proxy: EmbedderProxy = context.state.embedder_proxy.lock().unwrap().lock().clone();` instead here.